pallets/asset-index/src/lib.rs

```diff
@@ -327,8 +388,47 @@ pub mod pallet {
             Ok(())
         }
 
-        fn remove_asset(_: &T::AssetId) -> DispatchResult {
-            todo!();
+        /// Remove asset with provied units.
+        ///
+        /// * If is liquid asset, transfer the ownership to the provied recipient
+        /// * Remove asset with provied units directly if is SAFT
+        fn remove_asset(
+            asset_id: &T::AssetId,
+            units: &T::Balance,
+            recipient: Option<MultiLocation>,
```

Comment:
I'm beginning to think that naming `remove_asset` in combination with the recipient argument is too ambiguous. Depending on the assetid to recipient combination, this call acts either as
* a transfer, if the asset is liquid and the recipient is present
* a burn otherwise